lib/rubygems/ext/builder.rb

```diff
@@ -200,6 +212,7 @@ def build_extensions
 
     FileUtils.rm_f @spec.gem_build_complete_path
 
+    # FIXME: action at a distance: @ran_rake modified deep in build_extension(). - @duckinator
```

Comment:
is this still needed?

Reply:
I feel it is. This part of the codebase is very awkward to me &mdash; there's no reason to assume that `build_extensions` calls a method which calls a second method which _then_ modifies the variable that affects the loop. But it does.